test/framework/namespace_helpers.go

```diff
@@ -58,12 +59,14 @@ func CreateNamespace(ctx context.Context, input CreateNamespaceInput, intervals
 	if input.Name == "" {
 		input.Name = fmt.Sprintf("test-%s", util.RandomString(6))
 	}
-
 	ns := &corev1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: input.Name,
 		},
 	}
+	if input.Labels != nil {
+		ns.Labels = input.Labels
```

Comment:
nit: Might as well always set it in l.65

Setting something that is nil to nil is a no-op